### PR TITLE
v4.0.3

### PR DESCRIPTION
**Pull request by leofang**
**UPDATE**

This PR syncs with the upstream (conda-forge/openmpi-feedstock@9f390fb) and updates the version to v4.0.3.

A bit of context: a CUDA docker image is used in the CI because we built a CUDA-aware MPI in CF, which requires the presence of CUDA library at build time. 

However, at runtime we turned off this feature by default. To enable it, two conditions must be met: 1. the CUDA driver is present on the system (it's automatically searched and `dlopen`'d by Open MPI), and 2. the user explicitly sets an MCA parameter. This decision was made after a long discussion, to which I can link if anyone is interested 🙂

~~***DO NOT MERGE YET***~~
~~Trying to see if CF's cuda docker image is accessible in our workflow or not...~~

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,24 +13,31 @@ source:
 
 build:
   number: 0
-  skip: True  # [win or osx]
+  skip: true  # [win or osx or (linux64 and cuda_compiler_version != '9.2')]
```
Comment (mrakitin) [inline recipe/meta.yaml:16]: Ah, I guess we have such `True` issues in quite a few places.

Comment (leofang) [reply recipe/meta.yaml:16]: yeah I forgot: which one should we use and why?

Comment (mrakitin) [reply recipe/meta.yaml:16]: I think the yaml format accepts the lower-cased version, `true`. `True` came from the Python world.

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -2,7 +2,21 @@ channel_sources:
   - nsls2forge,defaults
 channel_targets:
   - nsls2forge main
-MACOSX_DEPLOYMENT_TARGET:
-  - "10.12"
-macos_min_version:
-  - "10.12"
```
Comment (leofang) [inline recipe/conda_build_config.yaml:8]: I removed this because we aren't building for OS X.

File: recipe/meta.yaml (hunk 1 above)
Comment (mrakitin) [reply recipe/meta.yaml:16]: Actually, it does not care: https://yaml.org/type/bool.html.

Comment (leofang) [reply recipe/meta.yaml:16]: Ah OK. I'll then leave it as is. (I copied it from upstream.)

**Comment by leofang** [comment]
PR description updated.

**Comment by leofang** [comment]
@mrakitin This is ready.

**Comment by mrakitin** [comment]
Just to confirm, does it use the internal machine for building? I can't see the building logs here.

**Comment by leofang** [comment]
No, this still uses Azure runners. Only the docker image is changed.

**Comment by leofang** [comment]
Log is here: https://dev.azure.com/nsls2forge/nsls2forge/_build/results?buildId=2422&view=results

**Review by mrakitin — APPROVED** [review]

**Pull request merged**